iso15118/secc/controller/evse_data.py

```diff
@@ -0,0 +1,126 @@
+    evse_target_reactive_power_l2: Optional[float] = None
+    evse_target_reactive_power_l3: Optional[float] = None
+    evse_present_active_power: Optional[float] = None  # Optional in AC CPD
+    evse_present_active_power_l2: Optional[float] = None  # Optional in AC CPD
+    evse_present_active_power_l3: Optional[float] = None  # Optional in AC CPD
+
+
+@dataclass
+class EVSEACLimits(Limits):
+    # 15118-2 AC CPD
+    evse_nominal_voltage: Optional[float] = None  # Also required for 15118-20 CPD
+    evse_max_current: Optional[float] = None  # Required
+
+    # 15118-20 AC CPD (Required)
+    evse_max_charge_power: Optional[float] = None
+    evse_min_charge_power: Optional[float] = None
+
+    # 15118-20 AC CPD (Optional)
+    evse_max_charge_power_l2: Optional[float] = None
+    evse_max_charge_power_l3: Optional[float] = None
+    evse_min_charge_power_l2: Optional[float] = None
+    evse_min_charge_power_l3: Optional[float] = None
+    evse_nominal_frequency: Optional[float] = None
+    max_power_asymmetry: Optional[float] = None
+    evse_power_ramp_limit: Optional[float] = None
+
+    evse_present_active_power: Optional[float] = None  # Optional in AC Scheduled CL
+    evse_present_active_power_l2: Optional[float] = None  # Optional in AC Scheduled CL
+    evse_present_active_power_l3: Optional[float] = None  # Optional in AC Scheduled CL
+
+
+@dataclass
+class EVSEACBPTLimits(Limits):
+    evse_max_discharge_power: Optional[float] = None  # Required
+    evse_max_discharge_power_l2: Optional[float] = None  # Optional
+    evse_max_discharge_power_l3: Optional[float] = None  # Optional
+
+    evse_min_discharge_power: Optional[float] = None  # Required
+    evse_min_discharge_power_l2: Optional[float] = None  # Optional
+    evse_min_discharge_power_l3: Optional[float] = None  # Optional
+
+
+@dataclass
+class EVSEDCCLLimits(Limits):
+    # Optional in 15118-20 DC Scheduled CL
+    evse_max_charge_power: Optional[float] = None  # Required in 15118-20 Dynamic CL
+    evse_min_charge_power: Optional[float] = None  # Required in 15118-20 Dynamic CL
+    evse_max_charge_current: Optional[float] = None  # Required in 15118-20 Dynamic CL
+    evse_max_voltage: Optional[float] = None  # Required in 15118-20 Dynamic CL
+
+    # Optional and present in 15118-20 DC BPT CL (Scheduled)
+    evse_max_discharge_power: Optional[float] = None  # Req in 15118-20 Dynamic BPT CL
+    evse_min_discharge_power: Optional[float] = None  # Req in 15118-20 Dynamic BPT CL
+    evse_max_discharge_current: Optional[float] = None  # Req in 15118-20 Dynamic BPT CL
+    evse_min_voltage: Optional[float] = None  # Required in 15118-20 Dynamic BPT CL
+
+
+@dataclass
+class EVSEDCBPTLimits(Limits):
+    # Required in 15118-20 DC BPT CPD
+    evse_max_discharge_power: Optional[float] = None
+    evse_min_discharge_power: Optional[float] = None
+    evse_max_discharge_current: Optional[float] = None
+    evse_min_discharge_current: Optional[float] = None
```

Comment:
same comment as before, cant we just have 1 class?

Reply:
As mentioned above, have changed the BPT version to inherit from the "non-bpt" one.